cypress/e2e/0-guild/0-create-guild.spec.ts

```diff
@@ -2,74 +2,212 @@ before(() => {
+
+      extraContact().type("username@example.com")
+      getContinueBtn().should("not.be.disabled")
+
+      cy.getByDataTest("removeContactBtn").should("be.visible").click()
+      extraContact().should("not.exist")
+
+      getContinueBtn().should("not.be.disabled")
+    })
+
+    it("requires wallet to access templates", () => {
+      fillCustomizeGuildForm()
+
+      cy.getByDataTest("wallet-selector-modal")
+        .should("be.visible")
+        .within(() => {
+          cy.get("button[aria-label='Close']").click()
+        })
+
+      cy.getByDataTest("wallet-selector-modal").should("be.visible")
+    })
   })
 
-  it("can create a guild without platform", () => {
-    cy.contains("Create guild without platform").click()
-    cy.get(".chakra-step [data-status='active'] div").should("contain.text", "2")
-
-    cy.contains("Start from scratch").click({ force: true })
-    cy.get(".chakra-step p[data-status='active']").should("contain.text", "Basic")
-    cy.contains("Growth").click({ force: true })
-    cy.get(".chakra-step p[data-status='active']").should("contain.text", "Growth")
-
-    cy.contains("Next").click()
-
-    cy.get("input[name='name']").focus().blur()
-    cy.get("input[name='name'] ~ .chakra-collapse")
-      .should("exist")
-      .contains("This field is required")
-    cy.getByDataTest("create-guild-button").should("be.disabled")
-    cy.get("input[name='name']").type(
-      `${Cypress.env("platformlessGuildName")} ${Cypress.env("RUN_ID")}`
-    )
-    cy.getByDataTest("create-guild-button").should("be.disabled")
-
-    cy.get("input[name='socialLinks.TWITTER']").focus().blur()
-    cy.get("input[name='socialLinks.TWITTER']")
-      .parent()
-      .siblings(".chakra-collapse")
-      .contains("This field is required")
-    cy.getByDataTest("create-guild-button").should("be.disabled")
-    cy.get("input[name='socialLinks.TWITTER']").type("guild.xyz")
-    cy.get("input[name='socialLinks.TWITTER']")
-      .parent()
-      .siblings(".chakra-collapse")
-      .contains("Invalid X URL")
-    cy.getByDataTest("create-guild-button").should("be.disabled")
-    cy.get("input[name='socialLinks.TWITTER']").clear().type("twitter.com/guildxyz")
-    cy.getByDataTest("create-guild-button").should("be.enabled")
-
-    cy.intercept("POST", `${Cypress.env("guildApiUrl")}/guilds`).as(
-      "createGuildRequest"
-    )
-
-    cy.getByDataTest("create-guild-button").click()
-
-    cy.wait("@createGuildRequest").its("response.statusCode").should("eq", 201)
+  context("with wallet", () => {
+    beforeEach(() => {
+      cy.clearIndexedDB()
+      cy.visit("/create-guild")
+      cy.connectWallet()
+      fillCustomizeGuildForm()
+    })
+
+    it("can select role templates", () => {
+      // Selecting the first role template
+      getContinueBtn().should("be.disabled")
+      cy.get('#role-checkbox[data-test^="selected-role-"]').should("not.exist")
+      cy.get('[data-test^="role-"]')
```

Comment:
I think you could extract this selector just like you did with the continue button ;)